app/lib/katello/util/path_with_substitutions.rb

```diff
@@ -12,12 +14,28 @@ def initialize(path, substitutions)
         @substitutions = substitutions
         @path = path
         @resolved = []
+        generate_substitutions_from_path if @substitutions.empty?
+      end
+
+      def split_path
+        @split ||= @path.split('/').reject(&:blank?)
```

Comment:
I might would lowercase this as well, just in case some upper case letters are used somewhere in the manifest